components/common-go/log/log_test.go

```diff
@@ -85,3 +88,59 @@ func TestFromBuffer(t *testing.T) {
 		}
 	}
 }
+
+func TestScrubFormatter(t *testing.T) {
+	logger := logrus.New()
+	logger.SetFormatter(&scrubFormatter{
+		delegate: &logrus.JSONFormatter{
+			TimestampFormat: time.RFC3339Nano,
+		},
+	})
+
+	var buffer bytes.Buffer
+	logger.SetOutput(&buffer)
+
+	logTime := time.Now()
+	logger.WithTime(logTime).WithField("workspaceID", "1234567890").Info("email: anton@gitpod.io")
+
+	expectation := fmt.Sprintf("{\"level\":\"info\",\"msg\":\"email: [redacted:md5:937db5302d87a73afd8cc15bd99c82e9:email]\",\"time\":\"%s\",\"workspaceID\":\"[redacted:md5:e807f1fcf82d132f9bb018ca6738a19f]\"}\n", logTime.Format(time.RFC3339Nano))
+	if diff := cmp.Diff(expectation, buffer.String()); diff != "" {
+		t.Errorf("unexpected result (-want +got):\n%s", diff)
+	}
+}
+
+/*
+30/05/2023: scrubbing takes almost twice as long because of relection and regex replacements
```

Comment:
@csweichel @mads-hartmann it doubles formatting time because of reflection/regex/hashing for simpler cases

To be fair we don't suppose to log often? Only in error case in production?